src/chat.rs

```diff
@@ -1729,50 +1744,123 @@ pub async fn marknoticed_all_chats(context: &Context) -> Result<(), Error> {
     Ok(())
 }
 
+/// Emits an event to update the time until the next local deletion.
+///
+/// This takes into account only per-chat timeouts, because global device
+/// timeouts are at least one hour long and deletion is triggered often enough
+/// by user actions.
+pub async fn update_autodelete_timeout(context: &Context) {
+    let autodelete_timestamp: Option<i64> = match context
+        .sql
+        .query_get_value_result(
+            "SELECT autodelete_timestamp \
+         FROM msgs \
+         WHERE autodelete_timestamp != 0 \
+           AND chat_id != ? \
+         ORDER BY autodelete_timestamp ASC \
+         LIMIT 1",
+            paramsv![DC_CHAT_ID_TRASH],
```

Comment:
We shouldn't, but if we do, the messages will be moved into trash.

The purpose of the check is not to prevent setting the timer on special chat, but to skip already deleted messages. Otherwise we also need to check for device chat and saved messages chat, which don't have special IDs.